goproxytest/proxy.go

```diff
@@ -96,10 +96,15 @@ func (srv *Server) readModList() error {
 	}
 	for _, info := range infos {
 		name := info.Name()
-		if !strings.HasSuffix(name, ".txt") && !info.IsDir() {
+		switch {
+		case strings.HasSuffix(name, ".txt"):
+			name = strings.TrimSuffix(name, ".txt")
+		case strings.HasSuffix(name, ".txtar"):
+			name = strings.TrimSuffix(name, ".txtar")
+		case info.IsDir():
```

Comment:
Why are we allowing a directory when we weren't before? Maybe worth a comment?

Reply:
we did continue the loop in the case of `info.IsDir()` before; see the deleted line. I essentially inverted the `!foo && !bar` with a `break` into a `foo || bar` with an `else` (the `default` case) which breaks.

one logical difference is that the old code would accept `module_version.txt/` as a directory, happily stripping the `.txt` suffix. I think that was a coding mistake, because that feature was never documented - the docs only refer to `module_version.txt` as a file, and `module_version/` as a directory.

Reply:
Sorry, I misread the previous logic. All LGTM.